packages/synapse-react-client/src/lib/containers/auth/TwoFactorEnrollmentForm.tsx

```diff
@@ -0,0 +1,265 @@
+import React, { useEffect, useRef, useState } from 'react'
+import {
+  Box,
+  BoxProps,
+  Button,
+  Divider,
+  Link,
+  Paper,
+  Stack,
+  styled,
+  Typography,
+} from '@mui/material'
+import { StyledOuterContainer } from '../../components/styled/LeftRightPanel'
+import { StyledComponent } from '@emotion/styled'
+import { TotpSecret } from '../../utils/synapseTypes/TotpSecret'
+import { useGetNotificationEmail } from '../../utils/hooks/SynapseAPI'
+import { toCanvas } from 'qrcode'
+import TextField from '../TextField'
+import { SynapseSpinner } from '../LoadingScreen'
+import FullWidthAlert from '../FullWidthAlert'
+import {
+  useFinishTwoFactorEnrollment,
+  useStartTwoFactorEnrollment,
+} from '../../utils/hooks/SynapseAPI/auth/useTwoFactorEnrollment'
+import TwoFactorSecretDialog from './TwoFactorSecretDialog'
+
+/**
+ * Returns a URL that can be used to generate a QR code that 2FA authenticator apps can interpret
+ * @param secret
+ * @param subject
+ */
+function toOtpAuthUrl(secret: TotpSecret, subject: string) {
+  return `otpauth://totp/Synapse:${subject}?secret=${secret.secret}&issuer=Sage%20Bionetworks&algorithm=${secret.alg}&digits=${secret.digits}&period=${secret.period}`
```

Comment:
Candidate for using a new BrandContext or just change to SageBionetworks when the account app becomes the only way to log in
```suggestion
  return `otpauth://totp/SageBionetworks:${subject}?secret=${secret.secret}&issuer=Sage%20Bionetworks&algorithm=${secret.alg}&digits=${secret.digits}&period=${secret.period}`
```
```suggestion
  return `otpauth://totp/MobileToolBox:${subject}?secret=${secret.secret}&issuer=Sage%20Bionetworks&algorithm=${secret.alg}&digits=${secret.digits}&period=${secret.period}`
```